### PR TITLE
retry updates of nodes in daemonset e2e

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

**Comment by quinton-hoole** [comment]
Ref: #14072

File: test/e2e/daemon_set.go [hunk 1]
```diff
@@ -61,7 +61,15 @@ func clearNodeLabels(c *client.Client) error {
 	for _, node := range nodeList.Items {
 		if len(node.Labels) != 0 {
 			node.Labels = map[string]string{}
-			newNode, err := nodeClient.Update(&node)
+			var newNode *api.Node
+			err = wait.Poll(5*time.Second, 30*time.Second, func() (bool, error) {
```
Comment (quinton-hoole) [inline test/e2e/daemon_set.go]: Please make these named constants at the top of this file.

File: test/e2e/daemon_set.go [hunk 2]
```diff
@@ -61,7 +61,15 @@ func clearNodeLabels(c *client.Client) error {
...
+				newNode, err = nodeClient.Update(&node)
+				if err != nil {
```
Comment (quinton-hoole) [inline test/e2e/daemon_set.go]: I think that you should be more specific about the error codes on which you retry?

File: test/e2e/daemon_set.go [hunk 3]
```diff
@@ -214,7 +222,15 @@ func testDaemonSets(f *Framework) {
 	nodeList, err := nodeClient.List(labels.Everything(), fields.Everything())
 	Expect(len(nodeList.Items)).To(BeNumerically(">", 0))
 	nodeList.Items[0].Labels = nodeSelector
-	newNode, err := nodeClient.Update(&nodeList.Items[0])
+	var newNode *api.Node
```
Comment (quinton-hoole) [inline test/e2e/daemon_set.go]: Same comments as above for this block of code.

File: test/e2e/daemon_set.go [hunk 4]
```diff
@@ -224,7 +240,14 @@ func testDaemonSets(f *Framework) {
 	newNode, err = nodeClient.Get(newNode.Name)
 	Expect(err).NotTo(HaveOccurred(), "error getting node")
 	newNode.Labels = map[string]string{}
-	newNode, err = nodeClient.Update(newNode)
+	err = wait.Poll(5*time.Second, 30*time.Second, func() (bool, error) {
```
Comment (quinton-hoole) [inline test/e2e/daemon_set.go]: Same comments as above for this block of code.

**Comment by quinton-hoole** [comment]
Minor comments, otherwise LGTM.

**Comment by alex-mohr** [comment]
@k8s-bot test this please

**Comment by mikedanese** [comment]
@quinton-hoole PTAL. Addressed all comments.

**Comment by quinton-hoole** [comment]
Perfect!  LGTM.

**Pull request merged**